classes/reviewForm/ReviewFormElementDAO.inc.php

```diff
@@ -242,7 +242,7 @@ public function getByReviewFormId($reviewFormId, $rangeInfo = null, $included =
             $rangeInfo
         );
 
-        return new DAOResultFactory($result, $this, '_fromRow');
+        return new DAOResultFactory($result, $this, '_fromRow', [], $this->getSql(), [(int) $reviewFormId]);
```

Comment:
The `$sql` and `$params` parameters to `DAOResultFactory::__construct` are sunsetted hacks for places where the code previously relied on ADODB's result counting tools and we haven't been able to remove/rewrite reliant code yet. I think you can keep this deprecated aspect localized to just this class by doing this instead:
```sql
diff --git a/classes/reviewForm/ReviewFormElementDAO.inc.php b/classes/reviewForm/ReviewFormElementDAO.inc.php
index 9343d0a73..50314140a 100644
--- a/classes/reviewForm/ReviewFormElementDAO.inc.php
+++ b/classes/reviewForm/ReviewFormElementDAO.inc.php
@@ -232,17 +232,17 @@ class ReviewFormElementDAO extends \PKP\db\DAO
     public function getByReviewFormId($reviewFormId, $rangeInfo = null, $included = null)
     {
         $result = $this->retrieveRange(
-            'SELECT *
+            $sql = 'SELECT *
                        FROM review_form_elements
                        WHERE review_form_id = ?
                        ' . ($included === true ? ' AND included = 1' : '') . '
                        ' . ($included === false ? ' AND included = 0' : '') . '
                        ORDER BY seq',
-            [(int) $reviewFormId],
+            $params = [(int) $reviewFormId],
             $rangeInfo
         );
 
-        return new DAOResultFactory($result, $this, '_fromRow');
+        return new DAOResultFactory($result, $this, '_fromRow', [], $sql, $params);
     }
 
     /**
```

Reply:
@asmecher yeah thats another way to do this but I thought may be we need to have a way get the generated sql for this functionality in other places also as even if this is deprecated , we can not get rid of it completely right now . But as the same time right now I don't know where we may need this approach again that I have implemented so changing the implementation as you have mentioned .